File: CppPractice/Project1/Project1/Leetcode326.cpp
#include <string>

#include <vector>

#include <iostream>

#include <stdlib.h>

#include <time.h>

#include <sstream>

#include <algorithm>

#include <map>

#include <bitset>

using namespace std;

class Solution {
public:
    bool isPowerOfThree(int n) {
        return check_power_3(n, 0);

    }
    bool check_power_3(int n, int e) {
        if (n==pow(3,e))
        {
            return true;

        }
        if (n<pow(3,e))
        {
            return false;
        }
        return check_power_3(n, e + 1);
    }

};

File: Pythonleetcode/Leetcode326.py
class Solution(object):
    def isPowerOfThree(self, n):
        if n<=0:
            return False
        s=''
        while n!=0:
            s=str(n%3)+s
            n=n//3
        print(s)

        if ('1' not in s[1:]) and ('2' not in s[1:]) and s[0]=='1':
            return True
        else: return  False



print(Solution().isPowerOfThree(9))